python_modules/dagster/dagster/core/execution/plan/inputs.py

```diff
@@ -486,13 +488,69 @@ def compute_version(
         pipeline_def: PipelineDefinition,
         resolved_run_config: ResolvedRunConfig,
     ) -> Optional[str]:
+        solid_config = resolved_run_config.solids.get(str(self.solid_handle))
+        config_data = solid_config.inputs.get(self.input_name) if solid_config else None
 
-        config_data = self.get_associated_config(resolved_run_config)
-        input_def = self.get_associated_input_def(pipeline_def)
-        dagster_type = input_def.dagster_type
+        solid_def = pipeline_def.get_solid(self.solid_handle)
+        dagster_type = solid_def.input_def_named(self.input_name).dagster_type
         return dagster_type.loader.compute_loaded_input_version(config_data)
 
 
+@whitelist_for_serdes
+class FromDirectInputValue(
+    NamedTuple(
+        "_FromDirectInputValue",
+        [("input_name", str)],
+    ),
+    StepInputSource,
+):
```

Comment:
just to leave festering in your brain - theres a scheme where we allow direct value passing in composition functions as well and this pointer would refer to like node_handles to grab the captured value off `Node` objects